src/BugsnagUnity/BugsnagSettingsObject.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using BugsnagUnity;
+using BugsnagUnity.Payload;
+using System;
+
+namespace BugsnagUnity.Editor
+{
+    public class BugsnagSettingsObject : ScriptableObject
+    {
+
+        public bool AutoDetectErrors = true;
+        public bool AutoTrackSessions = true;
+        public string ApiKey;
+        public string AppType;
+        public ulong AppHangThresholdMillis;
```

Comment:
I guess all these numeric types being used map through to the equivalent property on the config object?  They seem a little inconsistent, though not a functional issue.

E.g. why `ulong` for this but `long` for launch duration?  And `int` for breadcrumbs?  Int32.Max ms is 3 weeks so they all could be `int` I'd have thought.

Reply:
Many of the data types are dictated by the native notifiers that we have to pass the values to. But i'll schedule a ticket to look at all the data types we use as i'm sure some are not well chosen